src/NServiceBus.Core/InitializableEndpoint.cs

```diff
@@ -32,20 +28,35 @@ internal Configure(SettingsHolder settings, IContainer container, List<Action<IC
             this.container.RegisterSingleton<ReadOnlySettings>(settings);
         }
 
-        /// <summary>
-        ///     Provides access to the settings holder.
-        /// </summary>
-        public SettingsHolder Settings { get; }
+        public Task<IStartableEndpoint> Initialize()
+        {
+            WireUpConfigSectionOverrides();
+
+            var featureActivator = new FeatureActivator(settings);
+
+            container.RegisterSingleton(featureActivator);
+
+            ForAllTypes<Feature>(TypesToScan, t => featureActivator.Add(t.Construct<Feature>()));
+
+            ForAllTypes<IWantToRunWhenBusStartsAndStops>(TypesToScan, t => container.ConfigureComponent(t, DependencyLifecycle.InstancePerCall));
+
+            ActivateAndInvoke<IFinalizeConfiguration>(TypesToScan, t => t.Run(settings));
+
+            var featureStats = featureActivator.SetupFeatures(new FeatureConfigurationContext(settings, container, pipelineSettings));
 
-        /// <summary>
-        ///     Gets the builder.
-        /// </summary>
-        public IBuilder Builder { get; private set; }
+            pipelineConfiguration.RegisterBehaviorsInContainer(settings, container);
 
-        /// <summary>
-        ///     Returns types in assemblies found in the current directory.
-        /// </summary>
-        public IList<Type> TypesToScan => Settings.GetAvailableTypes();
+            container.RegisterSingleton(featureStats);
+
+            featureActivator.RegisterStartupTasks(container);
+
+            ReportFeatures(featureStats);
+
+            var startableEndpoint = new StartableEndpoint(settings, builder, featureActivator, pipelineConfiguration, startables);
```

Comment:
Is it correct that we are no longer doing anything async here? Where are the features executed? If we are no longer doing anything async and we don't execute features which might become async we could think about returning `IStartableEndpoint` instead of `Task< IStartableEndpoint> `